src/java/org/apache/cassandra/index/sai/plan/Expression.java

```diff
@@ -110,12 +110,14 @@ public boolean isEqualityOrRange()
     public Expression(ColumnContext columnContext)
     {
         this.context = columnContext;
-        this.analyzer = columnContext.getAnalyzer();
+        this.analyzer = columnContext.getQueryAnalyzer();
         this.validator = columnContext.getValidator();
     }
 
     public Expression add(Operator op, ByteBuffer value)
     {
+        if (value == null) throw new IllegalArgumentException();
```

Comment:
Wouldn't these be better as assertions? or better - `Preconditions.checkNotNull`

Reply:
Yeah or I can just remove them